### PR TITLE
Generate CRDs yamls from _types.go using operator-sdk

### PR DESCRIPTION
**Pull request by qinqon**
**What this PR does / why we need it**:
Let's use the operator-sdk gen crds command so we can annotate source code at _types.go
with schema and openapi specific tags.

**Special notes for your reviewer**:
Depends on https://github.com/kubevirt/cluster-network-addons-operator/pull/569

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by qinqon** [comment]
we have to fix the following 
```
2020/09/29 14:18:32 Go Version: go1.13
2020/09/29 14:18:32 Go OS/Arch: linux/amd64
2020/09/29 14:18:32 version of operator-sdk: v0.18.0
2020/09/29 14:18:32 version of cluster-network-addons-operator: 99.0.0
2020/09/29 14:18:33 registering Components
2020/09/29 14:18:34 starting the operator manager
2020/09/29 14:19:16 reconciling NetworkAddonsConfig
2020/09/29 14:19:16 OpenShift cluster network configuration resource has not been found: no matches for kind "Network" in version "operator.openshift.io/v1"
2020/09/29 14:19:16 starting render phase
2020/09/29 14:19:16 render phase done, rendered 44 objects
2020/09/29 14:19:16 starting rendering objects to delete phase
2020/09/29 14:19:16 could not set reference for (/v1, Kind=ConfigMap) cluster-network-addons/cluster-networks-addons-operator-applied-cluster: cross-namespace owner references are disallowed, owner's namespace default, obj's namespace cluster-network-addons
2020/09/29 14:19:16 Failed to get NetworkAddonsConfig in order emit event. NetworkAddonsConfig.networkaddonsoperator.network.kubevirt.io "cluster" not found
2020/09/29 14:19:16 Failed to get NetworkAddonsOperator "cluster" in order to update its State: NetworkAddonsConfig.networkaddonsoperator.network.kubevirt.io "cluster" not found
2020/09/29 14:19:16 Successfully updated status conditions
2020/09/29 14:19:16 reconciling NetworkAddonsConfig
I0929 14:19:17.896783       1 request.go:621] Throttling request took 1.033412011s, request: GET:https://10.96.0.1:443/apis/scheduling.k8s.io/v1beta1?timeout=32s
2020/09/29 14:19:18 OpenShift cluster network configuration resource has not been found: no matches for kind "Network" in version "operator.openshift.io/v1"
2020/09/29 14:19:18 starting render phase
2020/09/29 14:19:18 render phase done, rendered 44 objects
2020/09/29 14:19:18 starting rendering objects to delete phase
2020/09/29 14:19:18 could not set reference for (/v1, Kind=ConfigMap) cluster-network-addons/cluster-networks-addons-operator-applied-cluster: cross-namespace owner references are disallowed, owner's namespace default, obj's namespace cluster-network-addons
2020/09/29 14:19:18 Failed to get NetworkAddonsConfig in order emit event. NetworkAddonsConfig.networkaddonsoperator.network.kubevirt.io "cluster" not found
2020/09/29 14:19:18 Failed to get NetworkAddonsOperator "cluster" in order to update its State: NetworkAddonsConfig.networkaddonsoperator.network.kubevirt.io "cluster" not found
2020/09/29 14:19:18 Successfully updated status conditions
2020/09/29 14:19:18 reconciling NetworkAddonsConfig
2020/09/29 14:19:20 OpenShift cluster network configuration resource has not been found: no matches for kind "Network" in version "operator.openshift.io/v1"
2020/
```

File: pkg/apis/networkaddonsoperator/shared/networkaddonsconfig_types.go [hunk 1]
```diff
@@ -77,14 +77,3 @@ type Container struct {
 	Name       string `json:"name"`
 	Image      string `json:"image"`
 }
-
```
Comment (RamLavi) [inline pkg/apis/networkaddonsoperator/shared/networkaddonsconfig_types.go:80]: why is this shared.NetworkAddonsConfig removed?
why do we move it to `networkaddonsconfig_controller.go`?

**Review by RamLavi — CHANGES_REQUESTED** [review]
not sure why `NetworkAddonsConfig` is moved to networkaddonsconfig_controller.go

File: pkg/apis/networkaddonsoperator/shared/networkaddonsconfig_types.go (hunk 1 above)
Comment (qinqon) [reply pkg/apis/networkaddonsoperator/shared/networkaddonsconfig_types.go:80]: If we have it at pkg/apis the gen crds will genreate a "shared" version in the yaml

File: .gitignore [hunk 2]
```diff
@@ -15,12 +15,17 @@ _components/
...
+# Generated crds
+deploy/crds/
+
 # Cover file
 cover.out
 
 # Makefile fake targets
 gen-k8s
-gen-k8s-check
+gen-openapi
```
[the remark this answers is not in the document]
Comment (qinqon) [reply .gitignore:26]: Yep with `touch` so we prevent re-run if not needed.

**Comment by qinqon** [comment]
/hold

There is a lot of stuff going on I was too naif, @RamLavi I think is better if we just try to merge the CRD description PR even if we have some duplication there and we work this out without hurry it up.

**Comment by qinqon** [comment]
The differences from generated and current one.
[crd.diff.txt](https://github.com/kubevirt/cluster-network-addons-operator/files/5326111/crd.diff.txt)

**Pull request closed without merging**